packages/react-components/react-combobox/src/utils/ComboboxBase.types.ts

```diff
@@ -148,3 +151,12 @@ export type ComboboxBaseContextValues = {
   activeDescendant: ActiveDescendantContextValue;
   listbox: ListboxContextValue;
 };
+
+export type HighlightedOptionChangeData = {
+  previousOption: OptionValue | null | undefined;
+  nextOption: OptionValue | null | undefined;
+};
+
+export type HighlightedOptionProps = {
+  onHighlightedOptionChange?: (event: ActiveDescendantChangeEvent, data: HighlightedOptionChangeData) => void;
```

Comment:
Why is there both `onActiveDescendantChange` and `onHightlightedOptionChange`? I think we should just keept he former, since the latter sets wrong expectations for users

Reply:
I don't like it - but I don't think `onActiveDescendantChange` is understandable for the component users - active descendant is implementation detail, not something we should expose in the API as a prop name I think.